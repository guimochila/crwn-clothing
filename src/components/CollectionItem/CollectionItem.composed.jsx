import React from 'react';
import { Mutation } from 'react-apollo';

import CollectionItem from './CollectionItem';
import { ADD_ITEM_TO_CART } from '../../graphql/queries';

const CollectionItemComposed = props => (
  <Mutation mutation={ADD_ITEM_TO_CART} variables={{ item: props.item }}>
    {addItemToCart => (
      <CollectionItem {...props} addItem={item => addItemToCart(item)} />
    )}
  </Mutation>
);

export default CollectionItemComposed;
