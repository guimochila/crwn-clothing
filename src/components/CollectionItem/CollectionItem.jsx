import React from 'react';
import {
  CollectionItemContainer,
  CollectionFooterContainer,
  AddButton,
  BackgroundImage,
  NameContainer,
  PriceContainer,
} from './CollectionItem.styles';

function CollectionItem({ item, addItem }) {
  const { name, price, imageUrl } = item;

  return (
    <CollectionItemContainer>
      <BackgroundImage className="image" imageUrl={imageUrl} />
      <CollectionFooterContainer>
        <NameContainer>{name}</NameContainer>
        <PriceContainer>{price}</PriceContainer>
      </CollectionFooterContainer>
      <AddButton inverted onClick={() => addItem(item)}>
        Add to cart
      </AddButton>
    </CollectionItemContainer>
  );
}

export default CollectionItem;
