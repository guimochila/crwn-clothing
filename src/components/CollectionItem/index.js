import CollectionItemComposed from './CollectionItem.composed';

export default CollectionItemComposed;
