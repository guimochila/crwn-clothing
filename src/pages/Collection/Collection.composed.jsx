import React from 'react';
import { Query } from 'react-apollo';

import CollectionPage from './Collection';
import Spinner from '../../components/Spinner';
import { GET_COLLECTIONS_BY_TITLE } from '../../graphql/queries';

const CollectionPageComposed = ({ match }) => (
  <Query
    query={GET_COLLECTIONS_BY_TITLE}
    variables={{ title: match.params.collectionId }}
  >
    {({ loading, data: { getCollectionsByTitle } }) => {
      if (loading) return <Spinner />;

      return <CollectionPage collection={getCollectionsByTitle} />;
    }}
  </Query>
);

export default CollectionPageComposed;
