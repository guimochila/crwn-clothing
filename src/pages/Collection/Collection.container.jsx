import { connect } from 'react-redux';
import { compose } from 'redux';
import { createStructuredSelector } from 'reselect';

import CollectionPage from './Collection';
import WithSpinner from '../../components/WithSpinner';
import { selectIsCollectionsLoaded } from '../../store/shop/shop.selectors';

const mapStateToProps = createStructuredSelector({
  isLoading: state => !selectIsCollectionsLoaded(state),
});

const CollectionPageContainer = compose(
  connect(mapStateToProps),
  WithSpinner,
)(CollectionPage);

export default CollectionPageContainer;
