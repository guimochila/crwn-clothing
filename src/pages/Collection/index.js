import CollectionPage from './Collection';
import CollectionPageContainer from './Collection.container';

export { CollectionPage, CollectionPageContainer };
