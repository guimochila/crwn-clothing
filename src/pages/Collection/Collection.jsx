import React from 'react';
import { connect } from 'react-redux';

import {
  CollectionPageContainer,
  CollectionTitle,
  CollectionItemsContainer,
} from './Collection.styles';
import NotFound from '../NotFound';
import CollectionItem from '../../components/CollectionItem';
import { selectCollection } from '../../store/shop/shop.selectors';

function CollectionPage({ collection }) {
  if (!collection) {
    return <NotFound />;
  }

  const { title, items } = collection;

  return (
    <CollectionPageContainer>
      <CollectionTitle>{title}</CollectionTitle>
      <CollectionItemsContainer>
        {items.map(item => (
          <CollectionItem key={item.id} item={item} />
        ))}
      </CollectionItemsContainer>
    </CollectionPageContainer>
  );
}

const mapStateToProps = (state, ownProps) => ({
  collection: selectCollection(ownProps.match.params.collectionId)(state),
});

export default connect(mapStateToProps)(CollectionPage);
