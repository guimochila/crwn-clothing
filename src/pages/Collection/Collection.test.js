import React from 'react';
import { shallow } from 'enzyme';

import { CollectionPage } from './Collection';
import CollectionItem from '../../components/CollectionItem';

describe('Collection Page', () => {
  let wrapper;
  let mockItems = [{ id: 1 }, { id: 2 }, { id: 3 }];

  beforeEach(() => {
    const mockCollection = {
      items: mockItems,
      title: 'Testing',
    };
    wrapper = shallow(<CollectionPage collection={mockCollection} />);
  });

  test('should render Collection Component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should render the same number of CollectionItems as collection array', () => {
    expect(wrapper.find(CollectionItem).length).toBe(mockItems.length);
  });
});
