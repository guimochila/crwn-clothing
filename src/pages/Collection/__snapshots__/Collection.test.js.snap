// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collection Page should render Collection Component 1`] = `ShallowWrapper {}`;
